pinot-controller/src/main/java/com/linkedin/pinot/controller/util/TableRetentionValidator.java

```diff
@@ -0,0 +1,195 @@
+        continue;
+      }
+
+      // Get the retention config
+      SegmentsValidationAndRetentionConfig retentionConfig = getTableConfig(tableName).getValidationConfig();
+      if (retentionConfig == null) {
+        LOGGER.error("Table: {}, \"segmentsConfig\" field is missing in table config", tableName);
+        continue;
+      }
+      String segmentPushType = retentionConfig.getSegmentPushType();
+      if (segmentPushType == null) {
+        LOGGER.error("Table: {}, null push type", tableName);
+        continue;
+      } else if (segmentPushType.equalsIgnoreCase("REFRESH")) {
+        continue;
+      } else if (!segmentPushType.equalsIgnoreCase("APPEND")) {
+        LOGGER.error("Table: {}, invalid push type: {}", tableName, segmentPushType);
+        continue;
+      }
+
+      // APPEND use case
+
+      // Get time unit
+      String timeUnitString = retentionConfig.getRetentionTimeUnit();
+      TimeUnit timeUnit;
+      try {
+        timeUnit = TimeUnit.valueOf(timeUnitString.toUpperCase());
+      } catch (Exception e) {
+        LOGGER.error("Table: {}, invalid time unit: {}", tableName, timeUnitString);
+        continue;
+      }
+
+      // Get time duration in days
+      String timeValueString = retentionConfig.getRetentionTimeValue();
+      long durationInDays;
+      try {
+        durationInDays = timeUnit.toDays(Long.valueOf(timeValueString));
+      } catch (Exception e) {
+        LOGGER.error("Table: {}, invalid time value: {}", tableName, timeValueString);
+        continue;
+      }
+      if (durationInDays <= 0) {
+        LOGGER.error("Table: {}, invalid retention duration in days: {}", tableName, durationInDays);
+        continue;
+      }
+      if (durationInDays > _durationInDaysThreshold) {
+        LOGGER.warn("Table: {}, retention duration in days is too large: {}", tableName, durationInDays);
+      }
+
+      // Skip segments metadata check for realtime tables
+      if (tableName.endsWith("REALTIME")) {
+        continue;
+      }
+
+      // Check segments metadata (only for offline tables)
+      List<String> segmentNames = getSegmentNames(tableName);
+      if (segmentNames == null || segmentNames.isEmpty()) {
+        LOGGER.warn("Table: {}, no segment metadata in property store", tableName);
+        continue;
+      }
+
+      List<String> errorMessages = new ArrayList<>();
+      for (String segmentName : segmentNames) {
+        OfflineSegmentZKMetadata offlineSegmentMetadata = getOfflineSegmentMetadata(tableName, segmentName);
+        TimeUnit segmentTimeUnit = offlineSegmentMetadata.getTimeUnit();
+        if (segmentTimeUnit == null) {
+          errorMessages.add("Segment: " + segmentName + " has null time unit");
+          continue;
+        }
+        long startTimeInMillis = segmentTimeUnit.toMillis(offlineSegmentMetadata.getStartTime());
+        if (!TimeUtils.timeValueInValidRange(startTimeInMillis)) {
```

Comment:
In SegmentZKMetadata, we never use yymmdd format as start/end timestamp. We already parsed them when generating the timestamp.